src/Moq/Language/Flow/WhenPhrase.cs

```diff
@@ -38,13 +38,19 @@ public ISetup<T> Setup(Expression<Action<T>> expression)
 
 		public ISetupSetter<T, TProperty> SetupSet<TProperty>(Action<T> setterExpression)
 		{
-			var setup = Mock.SetupSet(mock, setterExpression, this.condition);
+			Guard.NotNull(setterExpression, nameof(setterExpression));
+			var expression = ExpressionReconstructor.Instance.ReconstructExpression(setterExpression);
+
+			var setup = Mock.SetupSet(mock, expression, this.condition);
```

Comment:
That kind of code is duplicated a lot in other overloads. If `ReconstructExpression` were not generic, this code pattern could be pushed down into the `Mock` class, and this fluent API method (as well as others) would turn into one-liners.